### ISSUE TITLE
[Bug report] Markdown lists cause navigation error while using a custom theme.

### ISSUE THREAD

**Issue by yuchanns** [vuepress/vuepress-next#688]
### Description

I have made a very simple custom theme demo according to the [guide of document](https://v2.vuepress.vuejs.org/advanced/theme.html#create-a-theme), which has two navbar on its `Layout.vue` component.

On the begining, everything goes well, I can navigate from **home page** to **about page** just by clicking the link. You can experience this by the [online demo](https://vuepress-test-9mtzx7efk-yuchanns.vercel.app/) I deployed on vercel.

Thing came to happen when I added a list on `README.md`. I clicked the link on navbar and the conent of page disappeared, and a TypeError threw out on the console shows `TypeError: Cannot read properties of null (reading 'insertBefore')`. You can see it on [this version](https://vuepress-test-fi3rf3xa6-yuchanns.vercel.app/) I deployed.

Is it a bug? Or one my misbehavior?

**This only happens on build mode.**

update: I found this happens with markdown table too. And when I first visit the about page, it will navigate correctly to home page and then I can navigate back to about page without problem.

### Steps to reproduce

1. Clone the reprodution repo, build it and deploy as a site.
2. Click the about link on navbar.

- Reproduction link / repo: https://github.com/yuchanns/vuepress-test

### Expected behavior

Navigate correctly to see about page.

### Screenshots

<img width="1917" alt="image" src="https://user-images.githubusercontent.com/25029451/153748756-e25769e5-2bfc-409a-aa51-107561f259cf.png">

### Environment info

- Browser: Chrome 98.0.4758.80 arm64
- Output of `vuepress info`:

```bash
 System:
    OS: Linux 5.10 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (8) arm64 unknown
    Memory: 26.83 GB / 31.30 GB
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v16.13.2/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
  Utilities:
    Git: 2.25.1 - /usr/bin/git
  Browsers:
    Chrome: Not Found
    Firefox: Not Found
  npmPackages:
    @vuepress/bundler-vite:  2.0.0-beta.35
    @vuepress/bundler-webpack: Not Found
    @vuepress/cli:  2.0.0-beta.35
    @vuepress/client:  2.0.0-beta.35
    @vuepress/core:  2.0.0-beta.35
    @vuepress/markdown:  2.0.0-beta.35
    @vuepress/plugin-active-header-links:  2.0.0-beta.35
    @vuepress/plugin-back-to-top:  2.0.0-beta.35
    @vuepress/plugin-container:  2.0.0-beta.35
    @vuepress/plugin-docsearch: Not Found
    @vuepress/plugin-external-link-icon:  2.0.0-beta.35
    @vuepress/plugin-git:  2.0.0-beta.35
    @vuepress/plugin-google-analytics: Not Found
    @vuepress/plugin-medium-zoom:  2.0.0-beta.35
    @vuepress/plugin-nprogress:  2.0.0-beta.35
    @vuepress/plugin-palette:  2.0.0-beta.35
    @vuepress/plugin-prismjs:  2.0.0-beta.35
    @vuepress/plugin-pwa: Not Found
    @vuepress/plugin-pwa-popup: Not Found
    @vuepress/plugin-register-components: Not Found
    @vuepress/plugin-search: Not Found
    @vuepress/plugin-shiki: Not Found
    @vuepress/plugin-theme-data:  2.0.0-beta.35
    @vuepress/plugin-toc: Not Found
    @vuepress/shared:  2.0.0-beta.35
    @vuepress/theme-default:  2.0.0-beta.35
    @vuepress/utils:  2.0.0-beta.35
    vue:  3.2.31
    vue-loader: Not Found
    vue-router:  4.0.12
    vuepress: ^2.0.0-beta.35 => 2.0.0-beta.35
    vuepress-vite:  2.0.0-beta.35
    vuepress-webpack: Not Found
```

**Comment by Mister-Hope** [comment]
The issue is an internal error trigger by vue, and I am not sure how it happens. So I will leave this to you.

**Comment by meteorlxy** [comment]
https://github.com/vuepress/vuepress-next/issues/897#issuecomment-1131151366

**Comment by jlooper** [comment]
I'm seeing this as well

**Comment by meteorlxy** [comment]
I hope this has been fixed in 2.0.0-beta.46. Let me know if you still meet similar issues.
